### ISSUE TITLE
Allow to use `assert` outside `pytest.raises` block

### ISSUE THREAD

**Issue by dandreevv** [wemake-services/wemake-python-styleguide#2543]
# Rule request

There is impossible to validate correctness of custom exception.

## Thesis

I propose to allow `assert` statement to use variable outside context manager.

```
with pytest.raises(CustomError) as exc_info:
    client.send_sms(sms)

    assert exc_info.value.http_status == expected_status
    assert exc_info.value.description == expected_description
```
Code above raise an error:
```
148:16   WPS441 Found control variable used after block: exc_info
  assert exc_info.value.http_status == expected_status
```

## Reasoning

It is typical use case for [pytest](https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions).

**Comment by blablatdinov** [comment]
@sobolevn what is your opinion on this issue?

**Comment by sobolevn** [comment]
I agree, thanks a lot! Will be included in `1.0.0`
